daemon/container_operations_unix.go

```diff
@@ -353,6 +355,20 @@ func killProcessDirectly(cntr *container.Container) error {
 				logrus.Debug(e)
 				return e
 			}
+
+			// In case there were some exceptions(e.g., state of zombie and D)
+			if system.IsProcessAlive(pid) {
+
+				// Since we can not kill a zombie pid, add zombie check here
+				isZombie, err := system.IsProcessZombie(pid)
+				if err != nil {
+					logrus.Warningf("Container %s state is invalid", stringid.TruncateID(cntr.ID))
+					return err
+				}
+				if isZombie {
+					return fmt.Errorf("container %s PID %d is zombie, can not be killed. Add --init flag is suggested", stringid.TruncateID(cntr.ID), pid)
```

Comment:
Can you

- use `errors.Errorf(` for this (to preserve the error stack)
- wrap it in a `errdefs.System(` ? (at least, I think an "internal server error" is the correct error type for this)
- phrasing of the suggestion could be slightly updated (reading it, I think it should be more clear that the `--init` needs to be used when _creating_ the container, and not when trying to kill them 😅), I copied some of the wording from the `--init` flag's description (better suggestions welcome)


```suggestion
					return errdefs.System(errors.Errorf("container %s PID %d is zombie and can not be killed. Use the --init option when creating containers to run an init inside the container that forwards signals and reaps processes", stringid.TruncateID(cntr.ID), pid)
```

Reply:
Sorry for my clumsy phrasing, I thought I made it as an old fashion Unix style(short and blurred) :joy: